content/blog/2016/2016-10-11-stage-lock-milestone.adoc

```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title: "Controlling the Flow with Stage, Lock, and Milestone"
+tags:
+- pipeline
+- newfeatures
+author: hrmpw
+---
+
+NOTE: This is a guest post by link:https://github.com/hrmpw[Patrick Wolf],
+Director of Product Management at link:http://cloudbees.com[CloudBees].
+
+Recently the Pipeline team began making several changes to improve the `stage` step and increase control of concurrent builds in Pipeline. Until now the `stage` step has been the catch-all for functionality related to the flow of builds through the Pipeline: grouping build steps into visualized stages, limiting concurrent builds, and discarding stale builds.
+
+In order to improve upon each of these areas independently we decided to break this functionality into discrete steps rather than push more and more features into an already packed `stage` step.
+
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Stage+Step+Plugin[stage] - the `stage` step remains but is now focused on grouping steps and providing boundaries for Pipeline segments.
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Lockable+Resources+Plugin[lock] - the `lock` step throttles the number concurrent builds in a defined section of the Pipeline.
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Milestone+Step+Plugin[milestone] - the `milestone` step automatically discards builds that will finish out of order and become stale.
+
+Separating these concerns into explicit, independent steps allows for much greater control of Pipelines and broadens the set of possible use cases.
+
+=== Stage
+
+The `stage` step is a primary building block in Pipeline, dividing the steps of a Pipeline into explicit units and helping to visualize the progress. The visualization plugin is called "Stage View" due to this function of `stage`.
```

Comment:
The second sentence reads weirdly to me.  You could perhaps remove it and end the first sentence like this:

> helping to visualize the progress, for example with the [Stage View Plugin][https://wiki...].